### PR TITLE
let docker choose the host port

### PR DESCRIPTION
**Pull request by rlgomes**
per discussion on #68

we want the ports to get auto assigned by docker which would remove any
port collision issues

also added a documentation section on `Development & Debugging` which
lets others how to figure out what port got assigned per container.

### CODE REVIEW

**Comment by rlgomes** [comment]
@demmer @mstemm @dmehra

File: examples/README.md [hunk 1]
```diff
@@ -201,3 +201,30 @@ mounts of paths below /tmp will refer to the VM and not your Mac/etc
 host. See
 [this github page](https://github.com/docker/compose/issues/1039) for
 more details.
+
+### Development & Debugging
+
+When making changes to existing example containers or simply wanting to debug if
+a specific **backend** contains the data that you're expecting you will have the
+need to connect to said **backend**. You can easily see which containers are
```
Comment (mstemm) [inline examples/README.md]: This sentence is a bit tough to parse. How about "You may want to connect directly to backend data sources to diagnose problems and debug juttle programs."?

Comment (mstemm) [inline examples/README.md]: Also, how about a sentence mentioning that the docker-compose files expose all the underlying data sources with randomly assigned host ports? That would set up why the docker-compose ps is necessary.

**Comment by mstemm** [comment]
For consistency's sake, can you also expose the elasticsearch port for the elasticsearch-nginx instance in dc-nginx-logs.yml, and the mysql port for the mysql instance in dc-aws-cloudwatch.yml ?

**Comment by rlgomes** [comment]
@mstemm sure I was thinking about and so it shall be.

**Comment by rlgomes** [comment]
@mstemm updated with all suggestions, looks good ?

**Comment by mstemm** [comment]
lgtm!

**Pull request merged**